backend/parser/listener/listener.go

```diff
@@ -314,7 +334,25 @@ func (s *searchListener[T]) appendRules(value string) {
 			}
 		}
 	} else if s.currentOp == "!=" {
-		if strings.Contains(value, "*") {
+		if strings.HasPrefix(value, "/") && strings.HasSuffix(value, "/") {
+			value = strings.Trim(value, "/")
+			if traceAttributeKey {
+				s.rules = append(s.rules, s.sb.Var(sqlbuilder.Buildf("NOT "+s.attributesColumn+"[%s] REGEXP %s", s.currentKey, value)))
+				s.ops = append(s.ops, &FilterOperation{
+					Key:      s.currentKey,
+					Column:   s.attributesColumn,
+					Operator: OperatorNotRegExp,
+					Values:   []string{value},
+				})
+			} else {
+				s.rules = append(s.rules, s.sb.Var(sqlbuilder.Buildf("NOT "+filterKey+" REGEXP %s", value)))
```

Comment:
This differs from the other `NOT`'s because the NOT must come before the filter key. I'm not sure what the s.ops is used for below, but not sure if this edge case will break that. I did not notice any regressions when testing

Reply:
I wonder if we can simplify the code here by treating the `!=` operator as `NOT (<subquery>)` in all cases. I think the logic isn't 100% correct at the moment, for example `environment!=(prod OR "on-prem")` is treated as `environment!=prod OR environment!="on-prem"`.

<img width="312" alt="Screen Shot 2024-02-26 at 9 37 31 AM" src="https://github.com/highlight/highlight/assets/86132398/be0fb004-af0c-434d-a385-a522122236aa">

Reply:
This is a good callout! I've been looking at this change, and it feels a little risky and unrelated to these changes? I created [this card](https://linear.app/highlight/issue/HIG-4394/fix-nested-=-queries) to tackle this work separately